src/v/storage/snapshot.cc

```diff
@@ -46,6 +47,16 @@ snapshot_manager::open_snapshot(ss::sstring filename) {
     });
 }
 
+ss::future<uint64_t> snapshot_manager::get_snapshot_size(ss::sstring filename) {
+    auto path = snapshot_path(filename);
+    return ss::file_exists(path.string()).then([path](bool exists) {
+        if (!exists) {
+            return ss::make_ready_future<uint64_t>(0);
+        }
+        return ss::file_size(path.string());
```

Comment:
you can avoid the race condition here by calling file_size directly and handling the enoent case.